examples/src/main/scala/example/federation/v2/FederatedApp.scala

```diff
@@ -16,10 +16,9 @@ object FederatedApp extends ZIOAppDefault {
     _           <-
       Server
         .serve(
-          Http
-            .collectHttp[Request] { case _ -> Root / "api" / "graphql" =>
-              ZHttpAdapter.makeHttpService(HttpInterpreter(interpreter))
-            }
+          Routes(
+            Method.ANY / "api" / "graphql" -> ZHttpAdapter.makeHttpService(HttpInterpreter(interpreter)).toHandler
```

Comment:
Should we return a `Handler` if we have to do `toHandler` to use it?

Reply:
The only thing about it is that I'm unsure whether there might be other methods of composition that I haven't figured out yet. E.g., the HttpApp is also convertible to Routes

Reply:
But we're never going to use it as `Routes` because it doesn't have any path in it, right?

Maybe make `makeHttpService` return a `Handler` for the most common case but keep a variant that returns the `HttpApp`?